code/modules/mod/modules/modules_general.dm

```diff
@@ -263,6 +271,14 @@
 	mod_color_overide = light_color
 	on_deactivation()
 
+/obj/item/mod/module/flashlight/extinguish_light(force)
+	. = ..()
+	on_deactivation(FALSE)
+	if(force)
```

Comment:
From what I can see, this means that the cooldown will only be activated with shadow demon grasps, and nothing like umbrae powers, shadow demon melee attacks, or from shadow cocoons. Just want to confirm this. Otherwise LGTM

Reply:
force should only be used for non electrical lights eg fires

Reply:
Noted in discord, I'm gonna remove the force handling here